weight.cpp

```diff
@@ -42,6 +43,33 @@ bool is_active(const weight &w)
  return w.is_active();
 }
 
+std::vector<double> convert_to_double
+  (const std::vector<weight> &weights)
+{
+  std::vector<double> double_vector;
+
+  for(size_t i = 0; i != weights.size(); i++)
+    {
+      double_vector.push_back(weights[i].get_weight());
+    }
+  return double_vector;
+}
+
+std::vector<double> convert_to_double_or_zero
+  (const std::vector<weight> &weights)
+{
+  std::vector<double> double_vector;
+
+  for(size_t i = 0; i != weights.size(); i++)
+    {
+      if(weights[i].is_active()==true)
+        double_vector.push_back(weights[i].get_weight());
+      else
+        double_vector.push_back(0);
+    }
```

Comment:
Same as what suggested for `convert_to_double()` above about setting the vector size in advance